### ISSUE TITLE
Unable to intercept cy.requests

### ISSUE THREAD

**Issue by symon-765** [cypress-io/cypress#25818]
### Current behavior

Currently when performing requests with the cy.request() command with an cy.intercept() that is setup to intercept requests of the same URL, the intercept is not actioned.

### Desired behavior

In this situation, I would expect cypress to intercept the cy.request command and allow all existing interception functionality to be usuable on the request that was made via cy.request()

### Test code to reproduce

```js
cy.intercept(/api/admin/learn").as('learn')

cy.request({
  url: "/api/admin/learn",
  method: "POST",
  body: body})
  
 cy.wait('@learn').then((interception} => {
 ...
}

Cypress states the 1st request for the route never occurred.
```

### Cypress Version

12.5.1

### Node version

v18.13.0

### Operating System

Windows 10 x64

### Debug Logs

```shell
Timed out retrying after 5000ms: cy.wait() timed out waiting 5000ms for the 1st request to the route: learn. No request ever occurred.
```

**Comment by flotwig** [comment]
Hey @symon-765, this is by design. `cy.request` is for sending HTTP requests without using `fetch` or `XMLHttpRequest`. Because you can receive the output of `cy.request` directly in `cy.then`, you don't need `cy.intercept` to do anything with the request or response. Just do this:

```ts
cy.request({
  url: "/api/admin/learn",
  method: "POST",
  body: body})
.then((response) => {
 ...
})
```

See also:
* https://docs.cypress.io/api/commands/request#Syntax
* https://docs.cypress.io/api/commands/request#cyintercept

**Comment by symon-765** [comment]
You're right, I can indeed recieve the output using .then(response), however, what if I want to change the response before it reaches my application to something else?

**Comment by flotwig** [comment]
@symon-765 in that case you would use `cy.intercept` to "intercept" the request being sent, like in this example: https://docs.cypress.io/api/commands/intercept#Intercepting-a-response

```ts
cy.intercept("/api/admin/learn", (req) => {
  req.continue(res => {
    // modify res
  })
})
```

`cy.request` sends an HTTP request, so you don't need to use it here at all.

**Comment by symon-765** [comment]
But this requires that I use the UI of my application to first get the API request to be sent, I would like to use cy.request() to send the request, and then alter the response before it hits my application. I assume this is not possible.

The cy.request() command is sending a request to the server, the request exists, it gets a response, so why shouldn't I be able to intercept that request that is made (just like any other request that is made by my application) and alter the response?

**Comment by flotwig** [comment]
> But this requires that I use the UI of my application to first get the API request to be sent, I would like to use cy.request() to send the request, and then alter the response before it hits my application. I assume this is not possible.

How would the response reach your application if you haven't used the UI to originate the request?
 
> The cy.request() command is sending a request to the server, the request exists, it gets a response, so why shouldn't I be able to intercept that request that is made (just like any other request that is made by my application) and alter the response?

If you want to send a request and intercept it, you could use `fetch` directly from your tests, but I don't see what purpose this serves. `cy.request` can't be intercepted since it bypasses `cy.intercept` by design.

**Comment by symon-765** [comment]
> How would the response reach your application if you haven't used the UI to originate the request?

The same way that cy.request() doesn't require me to use the UI to generate the request, I can send a cy.request at any point, and the request is made.

> If you want to send a request and intercept it, you could use `fetch` directly from your tests, but I don't see what purpose this serves. `cy.request` can't be intercepted since it bypasses `cy.intercept` by design.

Ok, I will try fetch, I guess I'm not explaining the scenario very well. thanks for help! :)

**Comment by nagash77** [comment]
@symon-765 please let us know how it goes and if you continue to have problems feel free to comment here.  Another great place to get help for things like this is the [Cypress Discord](https://discord.gg/cypress).  There are many knowledgable people there with a ton of Cypress experience.

**Comment by liebecampbellroberts** [comment]
Leaving this here for incase anyone else has this problem.

I had this problem ONLY when I deployed to Github Actions.  Locally it worked fine. 

The key to getting this to work for me is that the syntax for Spying and Stubbing are quite similar, its easy to not realise you are actually trying to make a call from GitHub Actions (which I suspect is blocking network requests) and so it replies :

`Timed out retrying after 5000ms: cy.wait() timed out waiting 5000ms for the 1st request to the route: learn. No request ever occurred.`

So I changed:
`
  cy.intercept({
    method: 'GET',
    url: loadingImage
  }).as('imageLoaded');
`
To:
`
cy.intercept(loadingImage, { fixture: `hero/${loadingImage}` }).as('imageLoaded');
`

Based on this:
<img width="607" alt="image" src="https://github.com/cypress-io/cypress/assets/12113864/ffd26971-54bd-408c-8263-27e950633fa5">
https://docs.cypress.io/api/commands/intercept#Stubbing-a-response

And the comments here explain the syntax:
<img width="394" alt="image" src="https://github.com/cypress-io/cypress/assets/12113864/4462a460-5ce5-415e-b160-d499c4ec5820">
https://docs.cypress.io/api/commands/intercept#Usage

Hope this saves you some time!
